core/src/test/java/ai/timefold/solver/core/impl/testdata/domain/shadow/multiplelistener/TestdataListMultipleShadowVariableEntity.java

```diff
@@ -0,0 +1,69 @@
+package ai.timefold.solver.core.impl.testdata.domain.shadow.multiplelistener;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import ai.timefold.solver.core.api.domain.entity.PlanningEntity;
+import ai.timefold.solver.core.api.domain.variable.PlanningListVariable;
+import ai.timefold.solver.core.impl.domain.entity.descriptor.EntityDescriptor;
+import ai.timefold.solver.core.impl.domain.variable.descriptor.ListVariableDescriptor;
+import ai.timefold.solver.core.impl.testdata.domain.TestdataObject;
+
+@PlanningEntity
+public class TestdataListMultipleShadowVariableEntity extends TestdataObject {
+
+    public static EntityDescriptor<TestdataListMultipleShadowVariableSolution> buildEntityDescriptor() {
+        return TestdataListMultipleShadowVariableSolution.buildSolutionDescriptor()
+                .findEntityDescriptorOrFail(TestdataListMultipleShadowVariableEntity.class);
+    }
+
+    public static ListVariableDescriptor<TestdataListMultipleShadowVariableSolution> buildVariableDescriptorForValueList() {
+        return (ListVariableDescriptor<TestdataListMultipleShadowVariableSolution>) buildEntityDescriptor()
+                .getGenuineVariableDescriptor("valueList");
+    }
+
+    public static TestdataListMultipleShadowVariableEntity createWithValues(String code,
+            TestdataListMultipleShadowVariableValue... values) {
+        // Set up shadow variables to preserve consistency.
+        return new TestdataListMultipleShadowVariableEntity(code, values).setUpShadowVariables();
+    }
+
+    TestdataListMultipleShadowVariableEntity setUpShadowVariables() {
```

Comment:
Why is this here, if the purpose of this is to test that the shadow vars are updated automatically?